### PR TITLE
Issue-1165: Migrate the Maven layout provider tests to use the new annotations (@ptirador)

### PR DESCRIPTION
**Pull request by ptirador**
Fixes for #1165.

Refactoring test cases to use new annotations.

* [x] `org.carlspring.strongbox.cron.jobs.CleanupExpiredArtifactsFromProxyRepositoriesCronJobTestIT`
* [x] `org.carlspring.strongbox.cron.jobs.ClearTrashCronJobFromMaven2RepositoryTestIT`
* [x] `org.carlspring.strongbox.cron.jobs.DownloadRemoteMavenIndexCronJobTestIT`
* [x] `org.carlspring.strongbox.cron.jobs.RebuildMavenIndexesCronJobTestIT`
* [x] `org.carlspring.strongbox.cron.jobs.RemoveTimestampedMavenSnapshotCronJobTestIT`

* [x] `org.carlspring.strongbox.storage.validation.version.MavenReleaseVersionValidatorTest`
* [x] `org.carlspring.strongbox.storage.validation.version.MavenSnapshotVersionValidatorTest` 

* [x] `org.carlspring.strongbox.storage.indexing.StrongboxIndexerTest`
* [x] `org.carlspring.strongbox.storage.indexing.RepositoryIndexerTest`

* [x] `org.carlspring.strongbox.dependency.snippet.MavenDependencyFormatterTest`

* [x] `org.carlspring.strongbox.providers.repository.MavenGroupRepositoryProviderTest` 

* [x] `org.carlspring.strongbox.storage.validation.artifactid.MavenArtifactIdLowercaseValidatorTest` 

* [x] `org.carlspring.strongbox.providers.search.MavenIndexerSearchProviderTest`

### CODE REVIEW

**Comment by fuss86** [comment]
@carlspring, maybe we could merge it ? Even if it does not contain full set of tests from maven provider module ? Refactorings of `org.carlspring.strongbox.storage.indexing.RepositoryIndexerTest` and `org.carlspring.strongbox.storage.indexing.StrongboxIndexerTest` would be helpful for me.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java [hunk 1]
```diff
@@ -53,11 +43,19 @@
 @SpringBootTest
 @ActiveProfiles(profiles = "test")
 @ContextConfiguration(classes = Maven2LayoutProviderTestConfig.class)
-@Execution(CONCURRENT)
+@Execution(SAME_THREAD)
```
Comment (fuss86) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: It stopped working concurrently ?

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java [hunk 2]
```diff
@@ -460,192 +294,141 @@ public void testGroupExcludes(@MavenRepository(repositoryId = "grpt-releases-tge
...
                                 Arrays.asList(new MutableRoutingRuleRepository(null, null)),
                                 ".*(com|org)/carlspring.metadata.*",
                                 RoutingRuleTypeEnum.DENY);
 
-        testDeny(repositoryReleases1Name, repositoryReleases2Name, repositoryGroupName);
+        testDeny(releases1.getId(), releases2.getId(), releasesGroup, (RepositoryPath) a2.normalize());
     }
 
-    @Test
-    public void deniedRoutingRuleShouldWorkIfRepositoryIsNotProvided(TestInfo testInfo)
```
Comment (fuss86) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: this test has been removed ?

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java (hunk 1 above)
Comment (ptirador) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: Yes, but only in Windows... Look at this [Jenkins Job](https://jenkins.carlspring.org/blue/organizations/jenkins/strongbox%2Fbuilds%2Fstrongbox/detail/PR-1370/12/pipeline). There were resource lock issues.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/cron/jobs/ClearTrashCronJobFromMaven2RepositoryTestIT.java [hunk 3]
```diff
@@ -66,43 +68,45 @@ public void testRemoveTrashInRepository(@MavenRepository(repositoryId = REPOSITO
...
+        RepositoryPath path = repositoryPathResolver.resolve(repository1, (RepositoryPath) artifact1.normalize());
         RepositoryPath trashPath = RepositoryFiles.trash(path);
         
         assertTrue(Files.exists(repositoryTrashPath), "There is no path to the repository trash!");
         assertFalse(Files.exists(trashPath), "The repository trash isn't empty!");
 
         RepositoryFiles.delete(path, false);
 
-        assertNotNull(Files.exists(repositoryTrashPath), "There is no path to the repository trash!");
+        Files.exists(repositoryTrashPath);
```
[the remark this answers is not in the document]
Comment (ptirador) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/cron/jobs/ClearTrashCronJobFromMaven2RepositoryTestIT.java]: Yes, my bad. It's beed added.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java (hunk 2 above)
Comment (ptirador) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: My mistake again... I've added it.

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java (hunk 1 above)
Comment (fuss86) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: Could you try enabling it one more time and let us know if it fails on Jenkins ?

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java (hunk 2 above)
Comment (fuss86) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: 👍

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java (hunk 1 above)
Comment (ptirador) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: Hah! Now it's working! Magic?

Comment (fuss86) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/MavenGroupRepositoryProviderTest.java]: I think it was Jenkins fault :) 👍

**Comment by carlspring** [comment]
Thank you very much, @ptirador!

**Pull request merged**